src/routes/privateRpc.ts

```diff
@@ -0,0 +1,56 @@
+import { Code, ConnectError, ConnectRouter } from '@connectrpc/connect';
+import { TypeOrmError } from '../errors';
+import { ArticlePost } from '../entity';
+import { generateShortId } from '../ids';
+import createOrGetConnection from '../db';
+import { isValidHttpUrl, standardizeURL } from '../common/links';
+import { baseRpcContext } from '../common/connectRpc';
+import { PostService } from '@dailydotdev/schema';
+
+export default function (router: ConnectRouter) {
+  router.rpc(PostService, PostService.methods.create, async (req, context) => {
+    if (!context.values.get(baseRpcContext).service) {
+      throw new ConnectError('unauthenticated', Code.Unauthenticated);
+    }
+
+    const con = await createOrGetConnection();
+
+    try {
+      req.url = standardizeURL(req.url);
+
+      if (!isValidHttpUrl(req.url)) {
```

Comment:
Yeah, per RFC `url` was required. I can make it optional but I think in that case we need to give the type in advance here so we now to validate or not validate URL depending on it?

Reply:
Collections have `source_id: "collection"`, could make exception for that condition. 
Or can add `type` variable, but would need to update schema definition for that.

Reply:
If we add type, would you be able to send it for all posts? I imagine sometimes the consumer would just want to get post_id and then figure out the type later?

Reply:
@denisb0 in case you missed it

Reply:
Looks like adding type brings in some extra complexity. But if you prefer to add `type` rather that checking `source_id == "collection" we can go that way. Currently yggdrasil can provide the type of the post it needs id for. Future consumers, if any, probably will also be aware of the type of content they want to create.

Reply:
@vpol do you have any thoughts on this?

Reply:
in case of collection source_id is artificial. If we can't find any other cases when sending a "type" is needed for any other "special" treatment on the api side, I would just stick to simple source_id check for now.

Reply:
Ok, in collection cases though it will require yggdrasil_id is sent.. So we can identify the post. @denisb0

Reply:
Technically there is also freeform/welcome post that does not have URL.

Reply:
Yeah, as long as API is the only source that generates that content.

Reply:
Added in a05637bd516cfad59eaad15e588cee007b428e83